core/src/main/java/hudson/Functions.java

```diff
@@ -233,6 +233,10 @@ public static void initPageVariables(JellyContext context) {
 
         // The path starts with a "/" character but does not end with a "/" character.
         context.setVariable("rootURL", rootURL);
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            jenkins.setCachedRootUrl(jenkins.getRootUrlFromRequest());
```

Comment:
Sorry guys, I don't understand the point here. What do you suggest? `setCachedRootURL(String)` can not be private because it needs to be called from here (and probably only here).